### DISCUSSION TITLE
Links' vertices with Ctrl key

### DISCUSSION THREAD

**Discussion by boligolov** [clientIO/joint#2196] [Q&A]
### Introduction

Hello for all!

I want to set vertices with `left mouse button` and `Ctrl` key pressed. What I have to do? 

I know, that I have to extend `linkTools.Vertices.VertexHandle` hadler and it has mention for some guards `if (this.options.guard(evt)) { return; }` in sources, but how to use them?

Thank you.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
You can extend the `onPointerDown` method of `VertexHandle` class.
```js
new joint.dia.ToolsView({
  tools: [new joint.linkTools.Vertices({
    handleClass: joint.linkTools.Vertices.VertexHandle.extend({
      onPointerDown: function(evt) {
        if (evt.button === 2 || !evt.ctrlKey) return;
        joint.linkTools.Vertices.VertexHandle.prototype.onPointerDown.apply(this, arguments);
      }
    })
  })]
}))
```
Here's an [example](https://jsfiddle.net/kumilingus/a5k1Lw9j/).

Analogically, you can override the `onPathPointerdown` method of the `Vertices` class.
